scrunch/datasets.py

```diff
@@ -998,20 +937,27 @@ class DatasetVariablesMixin(collections.Mapping):
 
     def __getitem__(self, item):
         # Check if the attribute corresponds to a variable alias
-        variable = self.resource.variables.by('alias').get(item)
+        variable = self._catalog.by('alias').get(item)
         if variable is None:
-            # Variable doesn't exists, must raise a ValueError
-            raise ValueError('Dataset %s has no variable with an alias %s' % (
-                self.name, item))
-        # Variable exists!, return the variable Instance
+            variable = self._catalog.by('name').get(item)
+            if variable is None:
+                # Variable doesn't exists, must raise a ValueError
+                raise ValueError('Entity %s has no (sub)variable with a name or alias %s' % (
+                    self.name, item))
         return Variable(variable, self)
 
     def _reload_variables(self):
         """
         Helper that takes care of updating self._vars on init and
         whenever the dataset adds a variable
         """
-        self._vars = self.resource.variables.index.items()
+        # this try except block is needed to update the 
+        # subvar or var catalogs once the Entity is modified
+        try:
+            self._catalog = self.resource.variables
+        except AttributeError:
+            self._catalog = self.resource.subvariables
```

Comment:
This pattern smells like you will be better off with a subclass that knows which sub catalog to use and just re-implement this method to use `self.resource.subvariables`